pyasic/miners/backends/epic.py

```diff
@@ -313,8 +317,22 @@ async def _get_hashboards(
                 hb_list[hb["Index"]].temp = hb["Temperature"]
         return hb_list
 
-    async def _is_mining(self, *args, **kwargs) -> Optional[bool]:
-        return None
+    async def _is_mining(self, web_summary, *args, **kwargs) -> Optional[bool]:
+        if web_summary is None:
+            try:
+                web_summary = await self.web.summary()
+            except APIError:
+                pass
+        if web_summary is not None:
+            try:
+                op_state = web_summary["Status"]["Operating State"]
+            except KeyError:
+                return None
+                pass
+        if op_state != "Idling":
+            return True
+        else:
+            return False
```

Comment:
I would just `return not op_state == "idling"` in your try block, then you can just pass on the key error and not have to do more parsing.

Reply:
done. much cleaner. thanks